content/learn/intermediate/data-types-2.md

```diff
@@ -1,178 +1,70 @@
-   \
-   Example :\
-   `{{$slice := cslice "This" 1 "Is" "Fun"}}{{$new := $slice.StringSlice}}`\
-   `{{joinStr "-" $new}}`\
-   \
-   Output :\
-   `This-Is-Fun`\
-   \
-   In the above snippet, initially a Slice is created with `cslice` called `$slice`. Then on invoking `.StringSlice` method on `$slice` , a new slice of type _\[]string_ containing only the _string_ datatype elements of _$slice_ is created and stored in `$new`. Then we using `joinStr` function to join the elements of the _\[]string_ slice into a single string with `-` as separator and the resultant string is output.
-
-{% hint style="info" %}
-Up until now we have been hearing about generic slices but it is interesting to note that we have already dealt with other types of slices than _templates.Slice_ datatype. Recall that `.CmdArgs` was defined as a slice of string arguments. Hence it is also a slice but of datatype _\[]String_.
-{% endhint %}
-
-{% hint style="info" %}
-Note that we have [already seen](https://learn.yagpdb.xyz/intermediate/control-flow-2#lets-go-deeper) the usage of the `seq` function in performing a repeated task a known number of times. However, did you know that seq actually also produces a slice. It is of type _\[]int_ and contains integers greater than equal to the first argument (start) upto one less than the second argument (stop).
-{% endhint %}
++++
+title = "Data Types 2"
+weight = 320
++++
+
+When you first started this course, you learned about [primitive data types](/learn/beginner/datatypes-1). In the
+previous chapter we name-dropped two more types: "slice", and "sdict" ("map"), both without much elaboration.
+In this chapter, we will explore these data types in more detail.
+
+## Slices
+
+Think of a slice as a list of items, that you can add to or remove from. You might have already heard of an array, which
+is a fixed-size list of items. A slice is like an array, but it can grow or shrink in size. This makes slices more
+powerful than a simple array.
+
+In custom commands, we can **c**reate a **slice** with the `cslice` function, like so:
+
+```go
+{{ $mySlice := cslice 1 2 3 "hello" 4 5 "good bye" }}
+```
+
+As you can see, the data types of the elements in a slice need not be equal, they can be anything. You already
+subconsciously used a slice with dictionaries as elements, when you created the `"fields"` slice for an embed in the
+previous chapter.
+
+For available operations on slices, please refer to [our template documentation][docs-slices].
+
+[docs-slices]: /docs/reference/templates/syntax-and-data/#templatesslice
+
+## Maps
+
+A map is an unordered collection of key-value pairs for fast lookups. In custom commands, we have two kinds of maps:
```

Comment:
A map is an table that associates values with keys. For example, a program that implements a reputation system might use a map from user IDs to reputation points internally. In custom commands, we have two kinds of maps: